app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -186,7 +187,7 @@ protected void initiateLoginAttempt(boolean restoreSession) {
             DevSessionRestorer.tryAutoLoginPasswordSave(uiController.getEnteredPasswordOrPin(), false);
         }
 
-        if (ResourceInstallUtils.isUpdateReadyToInstall()) {
+        if (ResourceInstallUtils.isUpdateReadyToInstall() && !UpdateActivity.isUpdateBlockedOnSync()) {

```

Comment:
@ctsims  Right now, we just restrict the auto updates here if a sync is needed and not auto trigger the sync afterwards the local login. Do you think it's the right behaviour or we should do a force sync once the user has logged in for these cases  ? Guessing we would not auto inititate the auto update (background) again anyway after login since the user is already inside the app.

Reply:
We do normally try to force syncs on Login, so it's not unusual for us to do so. Maybe the right behavior is to change the AutoSync behavior to `if ( autosync is pending OR update is blocked on sync)` which will mean that a sync attempt will trigger if needed every day but won't need to hijack the UX entirely the way that the pending update install will.?

One note - It occurs to me that we should also have some reporting mechanism that phones are stuck in this state - A device log event , and/or a

Reply:
(also fine to pursue these changes later, they definitely are not blocking)

Reply:
Makes sense, Made these changes in the last commit.